Mathlib/Analysis/NormedSpace/Multilinear/Basic.lean

```diff
@@ -91,25 +92,54 @@ We relate continuity of multilinear maps to the inequality `‖f m‖ ≤ C * 
 both directions. Along the way, we prove useful bounds on the difference `‖f m₁ - f m₂‖`.
 -/
 
-
 namespace MultilinearMap
 
 variable (f : MultilinearMap 𝕜 E G)
 
-/-- If a multilinear map in finitely many variables on normed spaces satisfies the inequality
-`‖f m‖ ≤ C * ∏ i, ‖m i‖` on a shell `ε i / ‖c i‖ < ‖m i‖ < ε i` for some positive numbers `ε i`
-and elements `c i : 𝕜`, `1 < ‖c i‖`, then it satisfies this inequality for all `m`. -/
-theorem bound_of_shell {ε : ι → ℝ} {C : ℝ} (hε : ∀ i, 0 < ε i) {c : ι → 𝕜} (hc : ∀ i, 1 < ‖c i‖)
+/-- If `f` is a continuous multilinear map in finitely many variables on `E` and `m` is an element
+of `(i : ι) → E i` such that one of the `m i` has norm `0`, then `f m` has norm `0`.
+
+Note that we cannot drop the continuity assumption because `f (m : Unit → E) = f (m ())`,
+where the domain has zero norm and the codomain has a nonzero norm
+does not satisfy this condition. -/
+lemma norm_map_coord_zero (hf : Continuous f) {m : (i : ι) → E i} {i : ι} (hi : ‖m i‖ = 0) :
+    ‖f m‖ = 0 := by
+  classical
+  have : Nonempty ι := ⟨i⟩
+  set m' : 𝕜 → (i : ι) → E i := fun ε : 𝕜 ↦ update (ε • m) i (ε⁻¹ ^ (Fintype.card ι) • ((ε • m) i))
+  have A : Tendsto m' (𝓝[≠] 0) (𝓝 0) := by
+    rw [← update_eq_self i (0 : (i : ι) → E i)]
```

Comment:
Should we take this opportunity to make these changes? Notice that the [`variable` block above](https://github.com/leanprover-community/mathlib4/pull/9700/files#diff-ae7e326549b303d39ecb126588c062ba023518dd4e0e052cdf019ba3b327217fR82) already uses `∀ i`. However, I noticed a lot of docstrings using `(i : ι) →` even though it's not valid Lean 3 syntax. So it seems mathport picked `∀ i` but people prefer `(i : ι) →`? It would be nice if we could agree on a uniform style ...
```suggestion
of `∀ i, E i` such that one of the `m i` has norm `0`, then `f m` has norm `0`.

Note that we cannot drop the continuity assumption because `f (m : Unit → E) = f (m ())`,
where the domain has zero norm and the codomain has a nonzero norm
does not satisfy this condition. -/
lemma norm_map_coord_zero (hf : Continuous f) {m : ∀ i, E i} {i : ι} (hi : ‖m i‖ = 0) :
    ‖f m‖ = 0 := by
  classical
  have : Nonempty ι := ⟨i⟩
  set m' : 𝕜 → ∀ i, E i := fun ε : 𝕜 ↦ update (ε • m) i (ε⁻¹ ^ (Fintype.card ι) • ((ε • m) i))
  have A : Tendsto m' (𝓝[≠] 0) (𝓝 0) := by
    rw [← update_eq_self i (0 : ∀ i, E i)]
```
(reposting since I didn't quote exactly the lines I intended to)

Reply:
Committed.
For me ` (i : ι) →E i` feels more natural, but it is just a matter of getting used to `∀ i, E i`.